### PR TITLE
Fix issues with `physics.set_shape()` when `Allow Dynamic Transforms` is on

### PR DESCRIPTION
**Pull request by AGulev**
Fix issues with `physics.set_shape()` which appears when `Allow Dynamic Transforms` option in `game.project` is on.

## PR checklist

* [ ] Code
	* [x] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: engine/physics/src/physics/physics_2d.cpp [hunk 1]
```diff
@@ -1044,6 +1044,7 @@ namespace dmPhysics
     {
         b2Shape* shape = (b2Shape*) _shape;
         shape->m_radius = radius * world->m_Context->m_Scale;
+        shape->m_creationScale = radius;
```
Comment (AGulev) [inline engine/physics/src/physics/physics_2d.cpp:1047]: Make sure we update `m_creationScale` (which is Defold specific modification of box2d) which used as a basic radius value for a  scale applying

File: engine/physics/src/physics/physics_2d.cpp [hunk 2]
```diff
@@ -1060,6 +1061,10 @@ namespace dmPhysics
             b2PolygonShape* polygon_shape = (b2PolygonShape*) _shape;
             float scale = world->m_Context->m_Scale;
             polygon_shape->SetAsBox(w * scale, h * scale, polygon_shape->m_centroid, angle);
+            for (int32 i = 0; i < polygon_shape->m_vertexCount; ++i)
+            {
+                polygon_shape->m_verticesOriginal[i] = polygon_shape->m_vertices[i];
```
Comment (AGulev) [inline engine/physics/src/physics/physics_2d.cpp:1066]: synchronize originals with the verticsies to make sure we can apply scale

File: engine/gamesys/src/gamesys/scripts/script_physics.cpp [hunk 3]
```diff
@@ -1398,6 +1398,14 @@ namespace dmGameSystem
         return 1;
     }
 
+    static void Physics_CheckIfShapeSizeValid(lua_State* L, float value)
+    {
+        if (value == 0.0f)
```
Comment (Jhonnyg) [inline engine/gamesys/src/gamesys/scripts/script_physics.cpp:1403]: I think you typically want to compare with an epsilon when comparing floats, there is no guarantee it is exactly 0.0f

File: engine/gamesys/src/gamesys/scripts/script_physics.cpp [hunk 4]
```diff
@@ -1398,6 +1398,14 @@ namespace dmGameSystem
...
+        {
+            luaL_error(L,"Shape size can't be 0");
```
Comment (Jhonnyg) [inline engine/gamesys/src/gamesys/scripts/script_physics.cpp:1405]: Hm, this is just checking a float and doesn't say anything about the actual shape, I would probably make the function take a shape info as an argument, and rename it to CheckShape or something (the script library usually names functions Checkxxx where xxx is the type). Something like this perhaps? (pseudo-code)

```
void CheckShape(ShapeInfo info)
{
#define check_val(type_str, v) \
	if (fabs(v) < epsilon) \
		lua_error("%s ");

	switch(info.type)
	{
	case box:
		check_val("box.x", info.boxdimensions[0]);
		check_val("box.y", info.boxdimensions[1]);
		break;
	case sphere:
		check_val("sphere.radius", info.radius);
		break;
	etc..
	}
#undef check_val
}
```

**Review by Jhonnyg — CHANGES_REQUESTED** [review]
Looks good! I think we can improve the error handling a bit, and perhaps add a test?

**Comment by AGulev** [comment]
@Jhonnyg I didn't add new function since for now this is the only place we use it, but I use marco with better error message like:
```
ERROR:SCRIPT: [string "src/gamesys/test/collision_object/get_set_err..."]:34: Shape 'shape_sphere_1' has invalid size '0' for 'diameter'
stack traceback:
  [C]:-1: in function set_shape
  [string "src/gamesys/test/collision_object/get_set_err..."]:34: in function <[string "src/gamesys/test/collision_object/get_set_err..."]:32>
  ```
  
  Also, I added a test

**Review by Jhonnyg — APPROVED** [review]

**Pull request merged**